### DISCUSSION TITLE
Реализация поиска адреса на карте с возможностью уточнения перетаскиванием метки

### DISCUSSION THREAD

**Discussion by jiexaspb** [yandex-maps-unofficial/vue-yandex-maps#215] [Q&A (2.x)] [answered]
Подскажите, как можно реализовать подобный вид, как это реализовано в Яндекс.Бизнес?
(показал на видео)

1. Поисковая строка с подсказки прямо поверх карт
2. Возможность уточнить точку путем перемещения метки

https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/4509131/97fc615a-9d65-4348-99e5-aa5a7306769a

**Comment by daniluk4000** [comment, marked as answer]
Здравствуйте!

> Поисковая строка с подсказки прямо поверх карт

Это придётся сделать своими силами. В помощь [API Яндекс Карт](https://yandex.ru/maps-api/docs) и компоненты по типу `<yandex-map-entity>`, `<yandex-map-control>` и пр.

> Возможность уточнить точку путем перемещения метки

У метки есть в `settings` параметр `draggable`, его можно использовать для вашей задачи

**Comment by jiexaspb** [comment]
@daniluk4000 Подскажите, а как я могу напрямую обращаться к объекту яндекс.карт?

Например, я хочу добавить вот такой код:

```
var searchControl = new ymaps.control.SearchControl({
    options: {
    // Будет производиться поиск по топонимам и организациям.
    provider: 'yandex#search'
   }
});
myMap.controls.add(searchControl);
```

**Reply by daniluk4000** [reply]
И да, и нет.

Обращаться напрямую можете через:
- ymaps3 константу
- v-model map (должен быть shallowRef) - в документации есть примеры

Однако SearchControl не существует в Картах 3.0. Ни у нас, ни у [Яндекса](https://yandex.ru/dev/jsapi30/doc/ru/dg/concepts/general). Поэтому и самостоятельно)

**Reply by jiexaspb** [reply]
@daniluk4000 странно, но они вот в документации дают пример https://yandex.ru/dev/jsapi-v2-1/doc/ru/v2-1/dg/concepts/geocoding/searchControl

**Reply by daniluk4000** [reply]
@jiexaspb это API версии 2.1. `vue-yandex-maps@next` работает на базе Карт 3.0, их доку я скинул в прошлом комменте

**Reply by jiexaspb** [reply]
@daniluk4000 я правильно понимаю, что ждать появление этого функционала от Яндекса в v3 не приходится, так как они поменяли концепцию в сторону "Сделайте сами"?

**Reply by daniluk4000** [reply]
@jiexaspb к сожалению, я не знаю, действительно ли Яндекс ввели этот концепт в Картах v3, но ответить на этот вопрос, как и на то, добавят ли они строку поиска, смогут только сами Яндекс. Роадмапов или чего-то подобного они не предоставляют, опять же, к большому сожалению.

Пока что приходится наблюдать то, что видим - новые Карты правда работают лучше, приятнее в использовании, но фич в них куда меньше, чем в 2.1.

**Comment by jiexaspb** [comment]
@daniluk4000 можно у вас попросить попросить пример кода, как получить доступ к ymaps.search функции в версии 3?
В документации у вас я не смог найти примера (честно искал и пытался разобраться)

**Reply by daniluk4000** [reply]
@jiexaspb действительно, доки нет, причина этому: у Яндекса её тоже нет)) 

Вообще, вам рекомендуется использовать отдельное API поиска по организациям: https://yandex.ru/dev/geosearch/doc/ru/

А документация, если её можно так назвать, доступна в типизации от Яндекса

![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/10450717/2ccb5b17-858f-4a61-8846-39233b4067b9)

И судя по всему, чтобы это API использовать, вам всё равно нужно создать ключ, как указано в описании их функции (можно вторым аргументом передать как класс new ymaps3.Config). Далее передать в первый аргумент *объект) поле text, в типизации есть и другие поля, можете почитать.

На самом деле, раз у Яндекса тут есть хоть какое-то описание, добавлю наверное их функции в документацию.

**Reply by daniluk4000** [reply]
Создал https://github.com/yandex-maps-unofficial/vue-yandex-maps/issues/217
